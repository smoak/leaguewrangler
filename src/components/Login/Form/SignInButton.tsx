import { Button } from '@mui/material';
import { FC } from 'react';

interface SignInButtonProps {
  readonly isDisabled: boolean;
  onClick(): void;
}

const SignInButton: FC<SignInButtonProps> = ({ isDisabled, onClick }) => {
  return (
    <Button
      type="submit"
      fullWidth
      variant="contained"
      onClick={(e) => {
        e.preventDefault();
        onClick();
      }}
      sx={{ mt: 3, mb: 2 }}
      disabled={isDisabled}
    >
      Sign in
    </Button>
  );
};

export default SignInButton;
