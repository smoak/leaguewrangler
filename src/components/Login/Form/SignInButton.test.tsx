import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import SignInButton from './SignInButton';

describe('SignInButton', () => {
  let onClick: jest.Mock;

  beforeEach(() => {
    onClick = jest.fn().mockName('onClick');
    render(<SignInButton className="className" isDisabled={false} onClick={onClick} />);
  });

  describe('when clicked', () => {
    beforeEach(async () => {
      const user = userEvent.setup();
      await user.click(screen.getByRole('button'));
    });

    it('calls the provided onClick function', () => {
      expect(onClick).toHaveBeenCalled();
    });
  });

  it('renders as expected', () => {
    expect(screen.getByRole('button', { name: 'Sign in' })).toBeInTheDocument();
  });
});
