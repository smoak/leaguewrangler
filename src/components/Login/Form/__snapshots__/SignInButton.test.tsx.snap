// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignInButton renders as expected 1`] = `
<WithStyles(ForwardRef(Button))
  className="className"
  color="primary"
  disabled={false}
  fullWidth={true}
  onClick={[Function]}
  type="submit"
  variant="contained"
>
  Sign in
</WithStyles(ForwardRef(Button))>
`;
